fe/fe-core/src/main/java/org/apache/doris/catalog/CatalogRecycleBin.java

```diff
@@ -811,18 +810,18 @@ public synchronized void recoverPartition(long dbId, OlapTable table, String par
             throw new DdlException("Can not recover partition[" + partitionName + "]. Partition item conflict.");
         }
 
-        // recover partition
+        // check if partition name exists
         Partition recoverPartition = recoverPartitionInfo.getPartition();
         Preconditions.checkState(recoverPartition.getName().equalsIgnoreCase(partitionName));
         if (!Strings.isNullOrEmpty(newPartitionName)) {
             if (table.checkPartitionNameExist(newPartitionName)) {
                 throw new DdlException("Partition name[" + newPartitionName + "] is already used");
             }
+            recoverPartition.setName(newPartitionName);
```

Comment:
before set new name,   check if table contains other partition with the same old partition name.  if not contains old partition name,  remove old partition name from table's nameToPartition ?